test/e2e/secrets_test.go

```diff
@@ -0,0 +1,188 @@
+package e2e
+
+import (
+	"bytes"
+	"context"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/pem"
+	"io/ioutil"
+	"log"
+	"testing"
+	"time"
+
+	framework "github.com/operator-framework/operator-sdk/pkg/test"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/crypto/ssh"
+	core "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kubeTypes "k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"github.com/openshift/windows-machine-config-operator/pkg/controller/secrets"
+)
+
+// getExpectedPublicKey returns the public key associated with the private key within the cloud-private-key secret
+func getExpectedPublicKey() (ssh.PublicKey, error) {
```

Comment:
nit: Having a struct for Pulickey,PrivateKey and corresponding hash in nodeConfig would have allowed you to reuse it and reduced the calls to CreatePubKeyHashAnnotation